scripts/setupTypeScriptRollup.js

```diff
@@ -0,0 +1,137 @@
+			svelteFile = svelteFile.replace("<script", '<script lang="ts"')
+			fs.writeFileSync(path.join(dir, elements[i].name), svelteFile)
+		} else if (elements[i].name.match(/^((?!json).)*js$/))
+			fs.renameSync(
+				path.join(dir, elements[i].name),
+				path.join(dir, elements[i].name).replace(".js", ".ts")
+			)
+	}
+}
+
+// Edit rollup config
+const rollupConfigPath = path.join(projectRoot, "rollup.config.js")
+let rollupConfig = fs.readFileSync(rollupConfigPath, "utf8")
+
+// Edit imports
+rollupConfig = rollupConfig.replace(
+	`'rollup-plugin-terser';`,
+	`'rollup-plugin-terser';
+import sveltePreprocess from 'svelte-preprocess';
+import typescript from '@rollup/plugin-typescript';`
+)
+
+// Edit inputs
+rollupConfig = rollupConfig.replace(
+	`input: config.client.input(),`,
+	`input: config.client.input().replace(/\.js$/, '.ts'),`
+)
+rollupConfig = rollupConfig.replace(
+	`input: config.server.input(),`,
+	`input: { server: config.server.input().server.replace(/\.js$/, ".ts") },`
+)
+rollupConfig = rollupConfig.replace(
+	`input: config.serviceworker.input(),`,
+	`input: config.serviceworker.input().replace(/\.js$/, '.ts'),`
+)
+
+// Add preprocess to the svelte config, this is tricky because there's no easy signifier.
+// Instead we look for 'hydratable: true,'
+rollupConfig = rollupConfig.replace(
+	new RegExp("hydratable: true,", "g"),
+	"hydratable: true,\n\t\t\t\tpreprocess: sveltePreprocess(),"
+)
+
+// Add TypeScript
+rollupConfig = rollupConfig.replace(
+	/commonjs\(\),?/g,
+	"commonjs(),\n\t\t\ttypescript({ sourceMap: dev }),"
+)
+
+// Save rollup config
+fs.writeFileSync(rollupConfigPath, rollupConfig)
+
+// Add TSConfig
+const tsconfig = `{
+	"extends": "@tsconfig/svelte/tsconfig.json",
+	"include": ["src/**/*", "src/node_modules/**/*"],
+	"exclude": ["node_modules/*", "__sapper__/*", "static/*"]
+}`
+const tsconfigPath = path.join(projectRoot, "tsconfig.json")
+fs.writeFileSync(tsconfigPath, tsconfig)
+
+// Delete this script, but not during testing
+if (!argv[2]) {
+	// Remove the script
+	fs.unlinkSync(path.join(__filename))
+
+	// Check for Mac's DS_store file, and if it's the only one left remove it
+	const remainingFiles = fs.readdirSync(path.join(__dirname))
+	if (remainingFiles.length === 1 && remainingFiles[0] === ".DS_store") {
+		fs.unlinkSync(path.join(__dirname, ".DS_store"))
+	}
+
+	// Check if the scripts folder is empty
+	if (fs.readdirSync(path.join(__dirname)).length === 0) {
+		// Remove the scripts folder
+		fs.rmdirSync(path.join(__dirname))
+	}
+}
+
+// Adds the extension recommendation
+fs.mkdirSync(path.join(projectRoot, ".vscode"))
+fs.writeFileSync(
+	path.join(projectRoot, ".vscode", "extensions.json"),
+	`{"recommendations": ["svelte.svelte-vscode"]}`
+)
+
+console.log("Converted to TypeScript.")
+
+if (fs.existsSync(path.join(projectRoot, "node_modules"))) {
+	console.log(
+		"\nYou will need to re-run your dependency manager to get started."
```

Comment:
I think we should just say "npm install" instead of "dependency manager" since that's used in all the Sapper docs in order to help the beginners. Folks that are advanced enough to have chosen yarn or something else will understand how to translate it

Reply:
Also conforming to: https://github.com/sveltejs/template/blob/master/scripts/setupTypeScript.js

But I can change it if you want!